cmd/kubeadm/app/preflight/checks.go

```diff
@@ -182,6 +182,38 @@ func (fac FileAvailableCheck) Check() (warnings, errors []error) {
 	return nil, errors
 }
 
+// FileContentCheck checks that the given file contains the string Content.
+type FileContentCheck struct {
+	Path    string
+	Content string
+}
+
+func (fcc FileContentCheck) Check() (warnings, errors []error) {
+	errors = []error{}
+	f, err := os.Open(fcc.Path)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%s does not exists", fcc.Path))
+		return nil, errors
+	}
+
+	lr := io.LimitReader(f, 1)
```

Comment:
Okay, first I thought that it was the **caller's problem** to ensure the file it want to check is not too large and therefore I suggested that.

What would be better indeed is to use a LimitReader with the `Content` byte array's length as the length argument.